src/main/java/io/castle/client/internal/utils/CastleContextBuilder.java

```diff
@@ -65,8 +65,18 @@ private void addHeaderValue(ArrayList<CastleHeader> castleHeadersList, String ke
         }
     }
 
+    /**
+     * Extract the clientId from the request.
+     * If __cid cookie setup the use that value, if not use header 'X-Castle-Client-Id', if not use empty string ''
+     * @param request
+     * @return
+     */
     private String setClientIdFromHttpServletRequest(HttpServletRequest request) {
         String cid = request.getHeader("X-Castle-Client-Id");
+        if( cid == null) {
+            //Default value is empty string, so client_id always will be send.
+            cid = "";
```

Comment:
This ensure that the client_id will be "" if no value is found. Tests check this.